spyder_remote_services/services/files/base.py

```diff
@@ -0,0 +1,436 @@
+
+import orjson
+from tornado.websocket import WebSocketHandler
+
+
+class FileWebSocketHandler(WebSocketHandler):
+    """
+    WebSocket handler for opening files and streaming data.
+
+    The protocol on message receive (JSON messages):
+      {
+        "method": "read", # "write", "seek", etc.  (required)
+        "kwargs": {...},  (optional)
+        "data": "<base64-encoded chunk>",  # all data is base64-encoded  (optional)
+      }
+
+    The protocol for sending data back to the client:
+      {
+        "status": 200,  # HTTP status code  (required)
+        "data": "<base64-encoded chunk>",  # response data if any  (optional)
+        "error": {"message": "error message",  (required)
+                  "traceback": ["line1", "line2", ...]  (optional)}  # if an error occurred  (optional)
+      }
+    """
+
+    LOCK_TIMEOUT = 100  # seconds
+
+    max_message_size = 5 * 1024 * 1024 * 1024  # 5 GB
+
+    __thread_lock = threading.Lock()
+
+    # ----------------------------------------------------------------
+    # Tornado WebSocket / Handler Hooks
+    # ----------------------------------------------------------------
+    async def open(self, path):
+        """Open file."""
+        self.mode = self.get_argument("mode", default="r")
+        self.atomic = self.get_argument("atomic", default="false") == "true"
+        lock = self.get_argument("lock", default="false") == "true"
+        self.encoding = self.get_argument("encoding", default="utf-8")
+
+        self.file: FileIO = None
+        try:
+            self.path = self._load_path(path)
+
+            if lock and not await self._acquire_lock(path):
+                self.close(
+                    1002,
+                    self._parse_json(
+                        HTTPStatus.LOCKED, message="File is locked"
+                    ),
+                )
+                return
+
+            self.file = await self._open_file()
+        except OSError as e:
+            self.log.warning("Error opening file", exc_info=e)
+            self.close(1002, self._parse_os_error(e))
+        except Exception as e:
+            self.log.exception("Error opening file")
+            self.close(1002, self._parse_error(e))
+        else:
+            await self._send_json(HTTPStatus.OK)
+
+    def on_close(self):
+        """Close file."""
+        if self.file is not None:
+            self._close_file()
+        if self.__locked:
+            self._release_lock()
+
+    async def on_message(self, raw_message):
+        """Handle incoming messages."""
+        self.log.debug("Received message: %s", raw_message)
+        try:
+            await self.handle_message(raw_message)
+        except Exception as e:
+            self.log.exception("Error handling message")
+            await self.write_message(self._parse_error(e), binary=True)
+
+    # ----------------------------------------------------------------
+    # Internal Helpers
+        method, kwargs = await self._parse_message(msg)
```

Comment:
```suggestion
    # ----------------------------------------------------------------
    async def handle_message(self, raw_message):
        method, kwargs = await self._parse_message(msg)
```

This was removed in the previous commit.